stories/components/SInputNumber.02_Addons.story.vue

```diff
@@ -0,0 +1,286 @@
+import SInputAddon from 'sefirot/components/SInputAddon.vue'
+import SInputNumber from 'sefirot/components/SInputNumber.vue'
+import { createDropdown } from 'sefirot/composables/Dropdown'
+import { useForm } from 'sefirot/composables/Form'
+import { maxValue, minValue, required } from 'sefirot/validation/rules'
+import { ref } from 'vue'
+
+interface Data {
+  amount1: number | null
+  amount2: number | null
+  amount3: number | null
+  lottery1: number | null
+  lottery2: number | null
+  currency: number | null
+}
+
+const { data, validation, init, validateAndNotify } = useForm<Data>({
+  data: {
+    amount1: null,
+    amount2: null,
+    amount3: null,
+    lottery1: null,
+    lottery2: null,
+    currency: null
+  },
+  rules: {
+    amount1: {
+      required: required()
+    },
+    amount2: {
+      required: required()
+    },
+    amount3: {
+      required: required()
+    },
+    lottery1: {
+      required: required(),
+      minValue: minValue(1_000_000),
+      maxValue: maxValue(9_999_999)
+    },
+    lottery2: {
+      required: required(),
+      minValue: minValue(1_000_000),
+      maxValue: maxValue(9_999_999)
+    },
+    currency: {
+      required: required()
+    }
+  }
+})
+
+function generate7DigitNumbers() {
+  const min = Math.ceil(1_000_000)
+  const max = Math.floor(9_999_999)
+
+  return Math.floor(Math.random() * (max - min + 1) + min)
+}
+
+const dropdown1SelectedValue = ref(1)
+const dropdown1 = createDropdown([
+  {
+    type: 'filter',
+    search: true,
+    selected: dropdown1SelectedValue,
+    options: [
+      { label: '¥', value: 1, onClick: (value: number) => dropdown1SelectedValue.value = value },
+      { label: '$', value: 2, onClick: (value: number) => dropdown1SelectedValue.value = value }
+    ]
+  }
+])
+
+const dropdown2 = createDropdown([
+  {
+    type: 'menu',
+    options: [
+      { label: 'Generate', onClick: () => data.amount3 = generate7DigitNumbers() },
+      { label: 'Reset', onClick: () => init() },
+      { label: 'Submit', onClick: () => validateAndNotify() }
+    ]
+  }
+])
+</script>
+
+<template>
+  <Board
+    title="Components / SInputNumber / 02. Addons"
+  >
+    <div class="root">
+      <div class="group">
+        <div class="title">Addon Text (Before)</div>
+        <div class="grid">
+          <SInputNumber
+            name="amount1"
+            label="Label"
+            info="Some helpful information."
+            note="Required"
+            help="Please fill in amount."
+            placeholder="1000"
+            :validation="validation.amount1"
+            v-model="data.amount1"
+            separator
+          >
+            <template #addon-before>
+              <SInputAddon label="¥" unclickable />
+            </template>
+          </SInputNumber>
+
+          <div class="actions">
+            <SButton size="small" mode="info" label="Submit" @click="validateAndNotify" />
+            <SButton size="small" mode="mute" label="Reset" @click="init" />
```

Comment:
It may be better that it is enough to have just one button section of `Submit` and `Reset`, although now there are for each `TextInput` component, since the function for them is common.

Reply:
I've removed these ones 👍